### PR TITLE
CDAT Migration: Prepare branch for merge to `main`

### LINKED ISSUE #853: [Refactor]: CDAT Migration Phase 3: Prepare for merge to `main`
### Is your feature request related to a problem?

Prepare for merge to `main`. Linked PR #885 has the tasks. 

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tomvothecoder**
## Description

- Closes #853

Summary of Changes

- Remove CDAT dependencies from conda env `.yml` files
  - Affected files: `conda-env/ci.yml`, `conda-env/dev.yml`
- Update documentation to replace CDAT and CDAT-based module references to Xarray/xCDAT
  - Affected files: `available-parameters.rst`, `install.rst`, `dev_guide/adding-new-diags-sets.rst`
- Remove CDAT-based derivation modules (`/derivations/`)
  - Affected files: `acme.py`, `default_regions.py`
- Remove CDAT-based driver modules (`/driver/utils/`)
  - Affected files: `climo.py`, `dataset.py`, `diurnal_cycle.py`, `general.py`
- Remove CDAT-based metrics functions
  - Affected files: `metrics/__init__.py`
- Replace `genutil.udunits` module with `cf_units.Unit` module
  - Affected files: `derivations/utils.py`
- Remove old script called `e3sm_diags_vars.py`
- Remove deprecated plot modules from `/cartopy` dir and delete `/cartopy` dir
- Refactor `_add_contour_plot()` to use `_get_output_dir()` and remove `parameter` arg + update all plot modules with this change
  - Affected files are all under `/plot`

Todo (later)
- Remove `auxiliary_tools/cdat_regression_testing` dir with regression testing material, which will no longer be needed at some point

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] New and existing unit tests pass with my changes (locally and CI/CD build)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: auxiliary_tools/aerosol_budget.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# NOTE: This module uses the deprecated e3sm_diags.driver.utils.dataset.Dataset
+# class, which was replaced by e3sm_diags.driver.utils.dataset_xr.Dataset.
```
Comment (tomvothecoder) [inline auxiliary_tools/aerosol_budget.py:2]: The only actual change in this file besides styling.

File: e3sm_diags/derivations/utils.py [hunk 2]
```diff
@@ -82,9 +82,9 @@ def convert_units(var: xr.DataArray, target_units: str):  # noqa: C901
     elif var.attrs["units"] in ["gN/m^2/day", "gP/m^2/day", "gC/m^2/day"]:
         pass
     else:
-        # FIXME: Replace genutil.udunits module.
-        temp = udunits(1.0, var.attrs["units"])
-        coeff, offset = temp.how(target_units)
+        temp = cf_units.Unit(var.attrs["units"])
+        target = cf_units.Unit(target_units)
+        coeff, offset = temp.convert(1, target), temp.convert(0, target)
```
Comment (tomvothecoder) [inline e3sm_diags/derivations/utils.py:87]: Replaces `genutil.udunits` with `cf_units.Unit` for unit conversion.

File: e3sm_diags/plot/utils.py [hunk 3]
```diff
@@ -246,6 +243,54 @@ def _add_contour_plot(
     return c_plot
 
 
+def _get_colormap(color_map: str):
```
Comment (tomvothecoder) [inline e3sm_diags/plot/utils.py:246]: Moved this function out of `/plot/__init__.py` and refactored.

File: e3sm_diags/plot/utils.py [hunk 4]
```diff
@@ -193,7 +193,6 @@ def _get_c_levels_and_norm(
 
 def _add_contour_plot(
     ax: matplotlib.axes.Axes,
-    parameter: CoreParameter,
```
Comment (tomvothecoder) [inline e3sm_diags/plot/utils.py:196]: Removed `parameter` arg because it is no longer needed.

**Review by tomvothecoder — COMMENTED** [review]
Hi @chengzhuzhang, this PR is ready for review. 

The summary of changes can be found in the PR's description. Thank you!

File: conda-env/dev.yml [hunk 5]
```diff
@@ -11,11 +11,9 @@ dependencies:
   - beautifulsoup4
   - cartopy >=0.17.0
   - cartopy_offlinedata
-  - cdms2 3.1.5
-  - cdutil 8.2.1
+  - cf-units
   - dask
   - esmpy >=8.4.0
-  - genutil 8.2.1
```
Comment (tomvothecoder) [inline conda-env/dev.yml:18]: No more CDAT and now Python 3.12+ compatible :)

**Review by chengzhuzhang — APPROVED** [review]
The code/doc changes to remove cdat looks good to me! Great work!
For the regression tests, I think we can re-purpose to include in our integration test, with the baseline change to the new `main` branch.

**Pull request merged**